pkg/operator/controllers/pullsecret/pullsecret_controller_test.go

```diff
@@ -163,12 +286,532 @@ func TestPullSecretReconciler(t *testing.T) {
 			}
 
 			if s.Type != corev1.SecretTypeDockerConfigJson {
-				t.Error(s.Type)
+				t.Errorf("Unexpected secret type: %s", s.Type)
 			}
 
 			if string(s.Data[corev1.DockerConfigJsonKey]) != tt.want {
-				t.Error(string(s.Data[corev1.DockerConfigJsonKey]))
+				t.Fatalf("Unexpected secret data.\ngot: %s\nwant: %s", string(s.Data[corev1.DockerConfigJsonKey]), tt.want)
+			}
+
+			cluster, err := r.arocli.AroV1alpha1().Clusters().Get(context.Background(), arov1alpha1.SingletonClusterName, metav1.GetOptions{})
+			if err != nil {
+				t.Fatal("Error found")
+			}
+
+		CONDITIONS:
+			for _, condition := range cluster.Status.Conditions {
+				for _, wantCondition := range tt.wantConditions {
+					if condition.Type == wantCondition.Type && condition.Status == wantCondition.Status {
+						continue CONDITIONS
+					}
+				}
+				t.Fatalf("Condition not found in cluster status.\ngot: %v\n want: %v", cluster.Status.Conditions, tt.wantConditions)
+			}
+		})
+	}
+}
+
+func TestCheckRHRegistryKeys(t *testing.T) {
+	test := []struct {
+		name     string
+		ps       map[string]string
+		wantKeys []string
+		wantErr  string
+	}{
+		{
+			name: "without rh key",
+			ps: map[string]string{
+				"arosvc.azrecr.io": "ZnJlZDplbnRlcg==",
+			},
+			wantKeys: []string{},
+		},
+		{
+			name: "with rh key",
+			ps: map[string]string{
+				"arosvc.azrecr.io":   "ZnJlZDplbnRlcg==",
+				"registry.redhat.io": "ZnJlZDplbnRlcg==",
+			},
+			wantKeys: []string{"registry.redhat.io"},
```

Comment:
This tests needs bit more attention:
```
{
			name: "with rh key",
			ps: map[string]string{
				"arosvc.azrecr.io":   "ZnJlZDplbnRlcg==",
				"registry.redhat.io": "ZnJlZDplbnRlcg==",
				"cloud.redhat.com":   "ZnJlZDplbnRlcg==",
			},
			wantKeys: []string{},
		},
```
and 
```
{
			name: "with rh key",
			ps: map[string]string{
				"arosvc.azrecr.io":   "ZnJlZDplbnRlcg==",
				"registry.redhat.io": "ZnJlZDplbnRlcg==",
				"cloud.redhat.com":   "ZnJlZDplbnRlcg==",
			},
			wantKeys: []string{"registry.redhat.io"},
		},
```
will pass too. In theory it is fine, but we never tests 2 keys, and test has false positives.

Reply:
O suspect conditions checking tests might have similar issue?